tests/test_containers.py

```diff
@@ -12,101 +12,204 @@
+    got_container_resgroup_names = {c.get_name():
+                                    c.get_resgroup().name for c in got_containers.values()}
     for expected_resgroup_name, container_names in mon_groups_relation.items():
         for container_name in container_names:
             if container_name in got_container_resgroup_names:
                 got_resgroup_name = got_container_resgroup_names.get(container_name)
                 assert got_resgroup_name == expected_resgroup_name
+
+
+ANY_METRIC_VALUE = 2
+
+
+@patch('owca.containers.Cgroup', spec=Cgroup,
+       get_measurements=Mock(return_value={'cgroup_metric__1': ANY_METRIC_VALUE}))
+@patch('owca.containers.PerfCounters', spec=PerfCounters,
+       get_measurements=Mock(return_value={'perf_event_metric__1': ANY_METRIC_VALUE}))
+def test_containerset_get_measurements(*args):
+    """Check whether summing of metrics for children containers are done properly.
+       Note: because we are mocking here classes from which measurements are read,
+       to calculate the proper value of ContainerSet we just need to multiple that
+       single value by count of subcontainers (here defined as N)."""
+    N = 3  # 3 subcontainers are created.
+    subcgroups_paths = ['/t1/c1', '/t1/c2', '/t1/c3']
+    containerset_ = container('/t1', subcgroups_paths, should_patch=False, rdt_enabled=True)
+
+    resgroup_mock = Mock(spec=ResGroup, get_measurements=Mock(return_value={'foo': 3}))
+    containerset_.set_resgroup(resgroup=resgroup_mock)
+
+    # Call the main function.
+    measurements = containerset_.get_measurements()
+
+    resgroup_mock.get_measurements.assert_called_once()
+    assert {'foo': 3, 'cgroup_metric__1': ANY_METRIC_VALUE*N,
+            'perf_event_metric__1': ANY_METRIC_VALUE*N} == measurements
+
+
+def smart_get_pids():
+    # Note: here List[int] is used instead of >>int<<
+    #   to pass mutable object (Integers are inmutable in Python).
+    calls_count = [0]
+
+    def fun() -> List[str]:
+        """Returns list of two consecutive integers. On first call
+           return [0,1], on second [2,3], and so forth."""
+        calls_count[0] += 2
+        return [str(calls_count[0] - 2), str(calls_count[0] - 1)]
+    return fun
+
+
+@patch('owca.containers.Cgroup')
+@patch('owca.containers.Container', spec=Container, get_pids=Mock(side_effect=smart_get_pids()))
+def test_containerset_get_pids(*args):
+    subcgroups_paths = ['/t1/c1', '/t1/c2', '/t1/c3']
+    containerset_ = container('/t1', subcgroups_paths)
+    # We expect 6 consecutive numbers starting from 0 - as there are 3 subcgroups paths
+    # for the container.
+    assert containerset_.get_pids() == [str(i) for i in range(6)]
+
+
+@patch('owca.containers.ResGroup.get_allocations', return_value={})
+@patch('owca.containers.Cgroup.get_allocations', return_value={})
+@patch('owca.containers.Container', spec=Container)
+def test_get_allocations(*mock):
```

Comment:
This test makes no sense, remove please.

Reply:
It has sense, it test if get_allocations do not use undefined variables or functions - before I use there unexisting variable self.cgroup instead of self._cgroup. It has little sense imho but it was suggested in first iteration of this PR review process.

Reply:
```suggestion
def test_container_get_allocations(*mock):
```
(it is unclear whether you test `Container` or `ContainerSet` to me).